### PR TITLE
Port the GUI from Agent5

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

Adds the python configuration gui back in.
Currently builds & runs (installation is next).

### CODE REVIEW

**Review by olivielpeau — COMMENTED** [review]
Made a few comments (mostly questions), overall feel free to merge this once the comments are addressed!

Also, do you have an installer built from this that I could try? :)

**Review by gmmeyer — DISMISSED** [review]
I will defer to your wisdom in porting all of this over for the GUI. I assume this was the best way to do it. I think this looks fine, as long as it works at this point we can trim it down later!

**Comment by masci** [comment]
Superseded by #669

**Pull request closed without merging**
